src/buildCommon.js

```diff
@@ -260,9 +260,19 @@ const makeOrd = function<NODETYPE: "spacing" | "mathord" | "textord">(
                                             options.fontShape);
             fontClasses = [fontOrFamily, options.fontWeight, options.fontShape];
         }
+
         if (lookupSymbol(value, fontName, mode).metrics) {
             return makeSymbol(value, fontName, mode, options,
                 classes.concat(fontClasses));
+        } else if (ligatures.hasOwnProperty(value) &&
+                   fontName.substr(0, 10) === "Typewriter") {
+            // Deconstruct ligatures in monospace fonts (\texttt, \tt).
```

Comment:
The reason why we deconstruct them instead of not constructing them in the first place is that we don't actually know what the current font is.  I wonder if we could form the ligatures in ordgroup.js instead.  That would avoid having to undo work.

Reply:
I thought about doing it in `buildExpression` and it might be possible, but it's worrying to me. For example, `-{}-` does not make a ligature, like `{-}-`. In the current parse tree, these can be distinguished from `--`, but with any optimization (such as what we do in MathML's buildExpression), they might not. It makes sense for ligatures to be at the lexing level, and doing it in the parser approximates this. I also don't imagine people use texttt too often, so it shouldn't cost much time. But I could also try it the other way.

Reply:
I read the TeXbook some about this.  One point is that `\lq\lq` makes a ligature, so it's good we do this after macro expansion, not in the lexer.

I believe one "right" way to do this would be if the parser already knew what font we were in.  This is difficult with the current separation between macros and functions, but maybe it could be done.  Definitely more effort though.

I think it'd be possible to do in `buildExpression`, but it'd have to be done twice, once for MathML and once for HTML.  Should I explore this further?